sdk/storage/azure-storage-blob/azure/storage/blob/_shared/base_client.py

```diff
@@ -212,7 +214,12 @@ def _batch_send(
         try:
             if response.status_code not in [202]:
                 raise HttpResponseError(response=response)
-            return response.parts()
+            parts = response.parts()
+            if raise_on_any_failure:
+                failures = [p for p in parts if p.status_code not in [202]]
+                if failures:
+                    raise HttpResponseError(message="There is a partial failure in the batch operation.")
```

Comment:
We not only want to to store the list of failures, we also want the list of successes too.
I think we just include the response:
```
error = HttpResponseError(message="There is a partial failure in the batch operation.", response=response)
error.failed_operations = failures
```
That way users can access the parts list:
```
try:
    ...
except HttpResponseError as e:
    all_responses = e.response.parts
```